### PR TITLE
New srv

### PR DESCRIPTION
**Pull request by privat**
New server based on bleeding edge popcorn techology

### CODE REVIEW

File: srv/tuto.nit [hunk 1]
```diff
@@ -0,0 +1,154 @@
+
```
Comment (privat) [inline srv/tuto.nit]: lit sens?

File: srv/web.nit [hunk 2]
```diff
@@ -0,0 +1,425 @@
...
+				status.solved = true
+			end
+
+			var score = 0
+			for t in tutorial.poset do
+				if session.solved(t) then score += 1
+			end
+			p.session_nav.add """
+			<li><a><span class="glyphicon glyphicon-star"></span> {{{score}}}/{{{tutorial.poset.length}}}</a></li>
+"""
```
Comment (privat) [inline srv/web.nit]: burkk

**Comment by privat** [comment]
ping @Morriar

File: srv/tuto.nit [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+module tuto
+
+import markdown
+import ini
+import poset
+
+# The main knowledge about lessons and exercises
+class Tutorial
+	# Dictionary of exercises by the internal name and the title id.
+	var exercise_by_name = new HashMap[String, Exercise]
```
Comment (Morriar) [inline srv/tuto.nit:12]: Why use both title_id and name?

File: srv/tuto.nit [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+
+	# Hierarchy of exercise dependencies
+	var poset = new POSet[Exercise]
+
+	# A poor man's database
+	var db = new ConfigTree("save.ini")
```
Comment (Morriar) [inline srv/tuto.nit:18]: Maybe use an attribute for the ini path?

**Comment by Morriar** [comment]
+1

Note sure about all the services stored into Session.

File: srv/tuto.nit (hunk 3 above)
Comment (privat) [inline srv/tuto.nit:12]: - name is the short crappy name used in the config files. It is used to locate file and build dependencies
- title_id is a striped version of the title, it is used in the url and other REST stuff.

Comment (Morriar) [inline srv/tuto.nit:12]: Maybe make two lists?

**Pull request merged**